backend/gateway-service/src/modules/user/user.controller.ts

```diff
@@ -1,30 +1,51 @@
 import {
   Body,
+  ClassSerializerInterceptor,
   Controller,
   Get,
   HttpCode,
   HttpStatus,
+  Inject,
+  Patch,
   Post,
   Query,
+  UseInterceptors,
 } from '@nestjs/common';
-import { UserService } from './user.service';
 import { ApiOkResponse, ApiTags } from '@nestjs/swagger';
-import { CreateUserDto } from './dto';
+import { GetCurrentUserId } from 'src/common/decorators';
+import { firstValueFrom } from 'rxjs';
+import { ClientProxy } from '@nestjs/microservices';
+import { UpdateUserDto, UsersResponseDto } from './dto';
+import { plainToInstance } from 'class-transformer';
 
 @ApiTags('users')
 @Controller('users')
+@UseInterceptors(ClassSerializerInterceptor)
 export class UserController {
-  constructor(private readonly userService: UserService) {}
+  constructor(
+    @Inject('USER_SERVICE') private readonly userClient: ClientProxy,
+  ) {}
 
-  @Get()
+  /* @Get()
   @HttpCode(HttpStatus.OK)
   getUserByEmail(@Query('email') email: string) {
     return this.userService.getUserByEmail(email);
+  } */
+
+  // Update question
+  @Patch('profile')
+  async updateProfile(@GetCurrentUserId() id: string, @Body() dto: UpdateUserDto): Promise<UsersResponseDto> {
+    const payload = { userId: id, updateUserDto: dto };
+    const updatedUser = await firstValueFrom(this.userClient.send({ cmd: 'update-user-profile' }, payload));
+    return plainToInstance(UsersResponseDto, updatedUser);
   }
 
-  // @Post()
-  // @HttpCode(HttpStatus.CREATED)
-  // createUser(@Body() createUserDto: CreateUserDto) {
-  //   return this.userService.createUser(createUserDto);
-  // }
+  @Get('profile')
+  @HttpCode(HttpStatus.OK)
+  async getUserDetails(@GetCurrentUserId() userId: string): Promise<UsersResponseDto> {
```

Comment:
Suggest to name this function to `getCurrentUserProfile` to emphasise that it is only to get current user. Anw it returns the current user only so js naming convention